### PR TITLE
Dashboard: Support Variables in "Filter by Name" Transformation

### LINKED ISSUE #42267: Template variables not work in Transform
**What happened**: Template variables not work in Transform, when I try to use "Filter by name" function, and fill in the Identifier.

**Environment**:
- Grafana version: 8.2.5

So is it the expected behavior?

### PR DESCRIPTION
**Pull request by Kirchen99**
**What this PR does / why we need it**:

This PR supports variables in "Filter by Name" transformation.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/42267

**Special notes for your reviewer**:

The `replace` function from the PR https://github.com/grafana/grafana/pull/44486 is used.

Discussion: https://github.com/grafana/grafana/discussions/50451

### CODE REVIEW

File: packages/grafana-data/src/transformations/transformers/filter.ts [hunk 1]
```diff
@@ -22,9 +22,25 @@ export const filterFieldsTransformer: DataTransformerInfo<FilterOptions> = {
...
+        ? replace(options.include?.options.pattern)
+        : options.include?.options.pattern;
+    }
+
+    if (typeof options.exclude?.options === 'string') {
+      options.exclude.options = replace ? replace(options.exclude?.options) : options.exclude?.options;
+    } else if (typeof options.exclude?.options?.pattern === 'string') {
+      options.exclude.options.pattern = replace
+        ? replace(options.exclude?.options.pattern)
+        : options.exclude?.options.pattern;
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-data/src/transformations/transformers/filter.ts:43]: This is slightly hard to read, could we change it to have the `replace` as a condition at the top?
```
    if (replace) {
      if (typeof options.include?.options === 'string') {
        options.include.options = replace(options.include?.options);
      } else if (typeof options.include?.options?.pattern === 'string') {
        options.include.options.pattern = replace(options.include?.options.pattern);
      }

      if (typeof options.exclude?.options === 'string') {
        options.exclude.options = replace(options.exclude?.options);
      } else if (typeof options.exclude?.options?.pattern === 'string') {
        options.exclude.options.pattern = replace(options.exclude?.options.pattern);
      }
    }
```

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Hi @Kirchen99 👋 Thanks for submitting this PR!
I have left a comment on it, to make it a bit easier to understand. Also, could we add a test to both of these transformations that were changed, to test with a `replace`? You can check the test of `calculateField` for inspiration 😄 

PS: I have added a comment to this ticket https://github.com/grafana/grafana/issues/46428  to make sure we reimplement most of this using `getTemplateSrv` directly when we move the transformers.

**Comment by Kirchen99** [comment]
> Hi @Kirchen99 👋 Thanks for submitting this PR! I have left a comment on it, to make it a bit easier to understand. Also, could we add a test to both of these transformations that were changed, to test with a `replace`? You can check the test of `calculateField` for inspiration 😄
> 
> PS: I have added a comment to this ticket #46428 to make sure we reimplement most of this using `getTemplateSrv` directly when we move the transformers.

Please have a look if the test I added is sufficient.

**Review by JoaoSilvaGrafana — APPROVED** [review]
Looks good, thanks for submitting this! 🎉

**Pull request merged**